packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -143,46 +175,63 @@ class WeatherScreen extends HookWidget {
                   }
                 }(),
               ),
-              child: Container(
-                constraints: const BoxConstraints.expand(),
-                child: SingleChildScrollView(
-                  physics: const BouncingScrollPhysics(),
-                  child: Column(
-                    children: [
-                      const MainInfoWidget(),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      SizedBox(
-                        height: MediaQuery.of(context).size.aspectRatio == 1.0
-                            ? 24.h
-                            : 16.h,
-                        child: const HourlyForecastsWidget(),
-                      ),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      const DailyForecastsWidget(),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      const AdditionalInfoWidget(),
-                    ],
-                  ),
-                ),
-              ),
+              child: fullWeatherState is w.Error &&
+                      fullWeatherState.fullWeather == null
+                  ? FailureWidget(
+                      failure: fullWeatherState.failure,
+                      onRetry: () async {
+                        await fullWeatherStateNotifier.loadFullWeather();
+                      },
+                    )
+                  : fullWeatherState.fullWeather == null
+                      ? Center(
+                          child: SpinKitSpinningLines(
+                            color: ClimaTheme.of(context).loadingIndicatorColor,
+                            size: 100.sp,
+                          ),
+                        )
```

Comment:
I honestly would prefer to just remove the dependency on `flutter_spinkit` and use what Flutter provides, but we can discuss that elsewhere.